main/templates/main/aidrequest_detail_hospital.html

```diff
@@ -17,13 +17,75 @@ <h1 class="text-3xl text-center font-bold">{{ object.title }}</h1>
 
 {% block content %}
 <div class="p-8">
-  {% include "includes/aidrequest_detail_fields.html" %}
+  <iframe name="hiddenFrame" class="hide"></iframe>
+  <form method="POST" target="hiddenFrame" id="id_myform">
+    {% csrf_token %}
 
-  <div class="p-3 mt-6">
-  {% url 'aidrequest_update' object.id as update %}
-  {% include "includes/button-link.html" with color="blue" href=update label="Edit" %}
-  {% url 'aidrequest_delete' object.id as delete %}
-  {% include "includes/button-link.html" with color="red" href=delete label="Delete" %}
-  </div>
+    {% include "includes/aidrequest_detail_fields.html" %}
+
+    <div class="p-3 mt-6">
+      {% url 'aidrequest_update' object.id as update %}
+      {% include "includes/button-link.html" with color="blue" href=update label="Edit" %}
+      {% url 'aidrequest_delete' object.id as delete %}
+      {% include "includes/button-link.html" with color="red" href=delete label="Delete" %}
+    </div>
+  </form>
 </div>
 {% endblock content %}
+
+{% block js %}
+<script>
+  setStatusOnClick('onload', '{{ object.status }}');
```

Comment:
This works for full page refreshes, but because we are using turbolinks for normal navigation this will be problematic.

I suggest running this on a dom page load, and on turbolinks:load event. See https://github.com/turbolinks/turbolinks